tools/offline_log_viewer/storage.py

```diff
@@ -289,8 +293,14 @@ class Segment:
     def __init__(self, path):
         self.path = path
 
+    def _term(self):
+        m = SEGMENT_NAME_PATTERN.match(os.path.basename(self.path))
```

Comment:
nit: maybe do this in the constructor and cache the term? It does't change, does it? Though I guess we'll create the iterator once anyway

Reply:
done